### PR TITLE
Thin3d shader cleanups

### PR DESCRIPTION
**Pull request by hrydgard**
This is prep for an upcoming pull request that adds tests that will be used to verify Adreno driver bugs.

Split these commits out to reduce the size of that pull request.

### CODE REVIEW

**Review by unknownbrackets — CHANGES_REQUESTED** [review]
Kinda separate question: do we need an `#if GL_FRAGMENT_PRECISION_HIGH` in StencilBufferGLES.cpp around `precision highp float;`?

Anyway, I think this requires changes to GPU (possibly not added / cherry-picked changes?) or it might cause errors.

-[Unknown]

File: ext/native/gfx_es2/gpu_features.cpp [hunk 1]
```diff
@@ -570,10 +570,23 @@ void SetGLCoreContext(bool flag) {
 }
 
 static const char *glsl_fragment_prelude =
+"#if __VERSION__ >= 130\n"
```
[the remark this answers is not in the document]
Comment (hrydgard) [reply ext/native/gfx_es2/gpu_features.cpp]: Yeah, this wasn't thought through enough. I got rid of all this and only kept the removal of the bytecode shader formats and the bug fix (which I already pushed to master but somehow shows up here anyway).

File: ext/native/thin3d/thin3d_vulkan.cpp [hunk 2]
```diff
@@ -211,10 +211,16 @@ class VKShaderModule : public ShaderModule {
...
+static const char *const vulkan_shader_preamble =
+"#version 450\n"
+"#extension GL_ARB_separate_shader_objects : enable\n"
+"#extension GL_ARB_shading_language_420pack : enable\n";
+
 bool VKShaderModule::Compile(VulkanContext *vulkan, ShaderLanguage language, const uint8_t *data, size_t size) {
 	vulkan_ = vulkan;
 	// We'll need this to free it later.
 	source_ = (const char *)data;
+	source_ = vulkan_shader_preamble + source_;
```
[the remark this answers is not in the document]
Comment (hrydgard) [reply ext/native/thin3d/thin3d_vulkan.cpp]: I reverted this.

**Comment by hrydgard** [comment]
Yes, we should probably add that ```#if GL_FRAGMENT_PRECISION_HIGH```...

I hate GLES.

**Review by unknownbrackets — APPROVED** [review]

**Pull request merged**